lib/utils/FrameView.hpp

```diff
@@ -0,0 +1,117 @@
+/*****************************************
+@file
+@brief Code for using the visBuffer formatted data.
+- VisMetadata
+- FrameView
+*****************************************/
+#ifndef FRAMEVIEW_HPP
+#define FRAMEVIEW_HPP
+
+#include "Hash.hpp"        // for Hash
+#include "buffer.h"        // for Buffer
+#include "chimeMetadata.h" // for chimeMetadata
+#include "dataset.hpp"     // for dset_id_t
+#include "visUtil.hpp"     // for cfloat
+
+#include "gsl-lite.hpp" // for span
+
+#include <set>      // for set
+#include <stdint.h> // for uint32_t, uint64_t, uint8_t
+#include <string>   // for string
+#include <time.h>   // for timespec
+#include <tuple>    // for tuple
+#include <utility>  // for pair
+
+
+/**
+ * @class FrameView
```

Comment:
I think you should document very clearly in here what the expected API of the specialisations of this class are, i.e. do you expect all implementations to have `calculate_buffer_layout` methods, and work in a certain way. There's some things you can't really enforce via the type system here, but you can enforce them by convention and documentation.